WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Crypto.ZeroKnowledge;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class AliceClient : BackgroundService, IAsyncDisposable
+	{
+		public Guid AliceId { get; }
+		public Guid RoundId { get; }
+		private ArenaClient ArenaClient { get; }
+		private IEnumerable<Coin> Coins { get; }
+
+		public AliceClient(Guid aliceId, Guid roundId, ArenaClient arenaClient, IEnumerable<Coin> coins)
+		{
+			AliceId = aliceId;
+			RoundId = roundId;
+			ArenaClient = arenaClient;
+			Coins = coins;
+		}
+
+		protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+		{
+			try
+			{
+				do
+				{
+					await Task.Delay(TimeSpan.FromSeconds(30), stoppingToken).ConfigureAwait(false);
+				}
+				while (!await ConfirmConnectionAsync().ConfigureAwait(false));
+			}
+			catch (Exception ex)
+			{
+				Logger.LogError($"Round ({RoundId}), Alice ({AliceId}): ConfirmConnection failed, reason: '{ex}'");
+			}
+		}
+
+		private async Task<bool> ConfirmConnectionAsync()
+		{
+			var inputWeight = 4 * Constants.P2wpkhInputVirtualSize;
```

Comment:
There is a constant for the `4` value too. This is something that I had to change to use `vbytes` anyway and this file will be iterated many times so, no need to do it in this PR but it could be good if it is changed.

Reply:
just leave it as 4, in my PRs i already started doing that and sticking to vsize, I think we can change one area at a time and the main remaining one is this, that can be its own separate PR and then the whole concept of WU will be gone from the codebase

Reply:
ok